src/passes/OptimizeInstructions.cpp

```diff
@@ -1426,6 +1450,8 @@ struct OptimizeInstructions
   // subtype of it. We will not remove a cast that would leave something that
   // would break that. If |requiredType| is not provided we will accept any type
   // there.
+  //
+  // See "notes on removing casts", above, for when this is safe to do.
   void skipCast(Expression*& input, Type requiredType = Type::anyref) {
```

Comment:
Do we need the `requiredType=anyref`? If the input is not a reference, it's not even valid, no?

Reply:
Yes, it wouldn't be valid. So `anyref` basically has no effect. That's what we want here - the default value means "no requirement on the type / accept any type there".